backends/bmv2/simple_switch/simpleSwitch.cpp

```diff
@@ -1173,6 +1173,7 @@ SimpleSwitchBackend::convert(const IR::ToplevelBlock* tlb) {
         // This is important later for conversion to JSON.
         new P4::ClonePathExpressions(),
         new P4::ClearTypeMap(typeMap),
+        new P4::TypeChecking(refMap, typeMap, true),
```

Comment:
This clears the typeMap and immediately recomputes it? Does this make sense?

Reply:
You are right, it should be removed.

Reply:
You are right, it should be removed.

Reply:
I don't know about this case, but whenever we rewrite the IR causing some  nodes to have a different type this is necessary.

Reply:
I think that ClearTypeMap call is needed. P4c have a bug without it for` bmv2/testdata/p4_16_samples/pvs-struct-3-bmv2.p4`: 
`Compiler Bug: /home/vpeschax/p4c/testdata/p4_16_samples/pvs-struct-3-bmv2.p4(17): Cannot find declaration for ternary @match(ternary) bit<16> field1; `